nexus/src/app/instance.rs

```diff
@@ -53,16 +54,80 @@ use uuid::Uuid;
 const MAX_KEYS_PER_INSTANCE: u32 = 8;
 
 impl super::Nexus {
+    pub async fn instance_lookup_id(
+        &self,
+        opctx: &OpContext,
+        instance_selector: params::InstanceSelector,
+    ) -> LookupResult<Uuid> {
+        match instance_selector {
+            params::InstanceSelector { instance: NameOrId::Id(id), .. } => {
+                // TODO: 400 if project or organization are present
+                Ok(id)
+            }
```

Comment:
Question (maybe for @davepacheco). Right now the ID is just a pass through, but if we did a lookup here it would ensure anytime a resource lookup happened you'd always know the user had read permissions. It's somewhat silly given that you're looking up something you potentially already have, but it would guarantee that after this point IDs are valid and at _least_ support read access.

Reply:
I chatted with folks in chat and my take away was that read access is the least permissible role. I went ahead and implemented this change which sort of guarantees the minimal access is read only.